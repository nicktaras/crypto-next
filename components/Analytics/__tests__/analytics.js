/* eslint-env jest */
import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Analytics from './../analytics.jsx';
jest.mock("react-ga")

// A simple snapshot test and child element length test to ensure that the analytics
// component can render child elements

describe('With Enzyme', () => {
  it('should have the child props', () => {
    const wrapper = shallow(<Analytics><p>Hello</p><p>World</p></Analytics>, {});
    expect(wrapper.find('p')).toHaveLength(2);
  })
})

describe('With Snapshot Testing', () => {
  it('should have rendered the child elements', () => {
    const component = renderer.create(<Analytics><p>Hello World</p></Analytics>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
