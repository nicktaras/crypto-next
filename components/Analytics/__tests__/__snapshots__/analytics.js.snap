// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should have rendered the child elements 1`] = `
<div>
  <p>
    Hello World
  </p>
</div>
`;
